docs/upgrade.md

```diff
@@ -6,164 +6,174 @@ sidebar_label: Upgrade
 
-Perform the following actions if you performed step 4 earlier.
+**Step-2 Upgrade OpenEBS old volumes created with 0.5.3 or0.5.4**
 
-- Access the Grafana dashboard at http://*NodeIP*:32515.
-- Add the Prometheus data source by specifying URL as http://*NodeIP*:32514.
-- Once the data source is validated, import the dashboard JSON from the <https://raw.githubusercontent.com/openebs/openebs/master/k8s/openebs-pg-dashboard.json> URL.
-- Access the volume statistics by selecting the volume name *(pvc-*)* in the OpenEBS Volume dashboard.
+Even after the OpenEBS Operator has been upgraded to 0.6, the old volumes will continue to work with 0.5.3 or 0.5.4. Each of the volumes should be upgraded (one at a time) to 0.6, using the steps provided below.
 
-**Note:** For new applications, select a newly created storage-class that has monitoring enabled to automatically start viewing metrics.
+**Note:** There has been a change in the way OpenEBS Controller Pods communicate with the Replica Pods. So, it is recommended to schedule a downtime for the application using the OpenEBS PV while performing this upgrade. Also, make sure you have taken a backup of the data before starting the below upgrade procedure.
 
-Detailed steps for supported upgrade paths are mentioned in the following sections.
+**Download the upgrade scripts**
 
-### **Upgrade from 0.4.0 to 0.5.0**
+You can get the requied scripts to your working directory using the following commands.
+```
+wget https://raw.githubusercontent.com/kmova/openebs/599e062a2251d2c16bca59aeac4a7ed77e445e6e/k8s/upgrades/0.5.x-0.6.0/controller.patch.tpl.yml
+wget https://raw.githubusercontent.com/kmova/openebs/599e062a2251d2c16bca59aeac4a7ed77e445e6e/k8s/upgrades/0.5.x-0.6.0/oebs_update.sh
+wget https://raw.githubusercontent.com/kmova/openebs/599e062a2251d2c16bca59aeac4a7ed77e445e6e/k8s/upgrades/0.5.x-0.6.0/patch-strategy-recreate.json
+wget https://raw.githubusercontent.com/kmova/openebs/599e062a2251d2c16bca59aeac4a7ed77e445e6e/k8s/upgrades/0.5.x-0.6.0/replica.patch.tpl.yml
+```
 
-It is possible to upgrade your OpenEBS volume from 0.4.0 to 0.5.0 by following the steps mentioned above. Detailed steps are mentioned here (<https://github.com/openebs/openebs/releases/tag/v0.5.0>). The README ensures better understanding of the change log. Limitations of the latest version are available at (https://github.com/openebs/openebs/blob/master/k8s/upgrades/0.4.0-0.5.0/README.md).
+In 0.5.x releases, when a replica is shutdown, it will get rescheduled to another availble node in the cluster and start copying the data from the other replicas. This is not a desired behaviour during upgrades, which will create new replica's as part of the rolling-upgrade. To pin the replicas or force them to the nodes where the data is already present, starting with 0.6 - we use the concept of nodeSelector and Tolerations that will make sure replica's are not moved on node or pod delete operations.
```

Comment:
and starts
which will create = as it creates new replicas due to rolling-upgrade
replica's = replicas
0.6, we use